### ISSUE TITLE
when I used the ImageDataGenerator,I encountered a error

### ISSUE THREAD

**Issue by hnsywangxin** [fizyr/keras-retinanet#186]
The parameter of ImageDataGenerator is :
```
       horizontal_flip=False,
       featurewise_center=True,
        featurewise_std_normalization=True,
        rotation_range=10,
        width_shift_range=0.1,
        height_shift_range=0.1
```

and the error is :min() arg is an empty sequence.
the error is from "keras_retinanet/utils/image.py", line 80, in random_transform
    boxes[index, 0] = float(min(j))"
thank you

**Comment by LaurensHagendoorn** [comment]
Jup, this can happen lot's basically a bounding box is augmented out of the image, and it cannot find it anymore, either lower your augmentation, or do something nasty like this: (or both)

`    while not finished:
        boxes=boxes_in
        if initseed is None:
            seed = np.random.randint(10000)
        else:
            seed = initseed

        image = image_data_generator.random_transform(image_in, seed=seed)

        # set fill mode so that masks are not enlarged
        fill_mode = image_data_generator.fill_mode
        image_data_generator.fill_mode = 'constant'

        for index in range(boxes_in.shape[0]):
            # generate box mask and randomly transform it
            mask = np.zeros_like(image, dtype=np.uint8)
            b = boxes_in[index, :4].astype(int)

            assert(b[0] < b[2] and b[1] < b[3]), 'Annotations contain invalid box: {}'.format(b)
            assert(b[2] <= image.shape[1] and b[3] <= image.shape[0]), 'Annotation ({}) is outside of image shape ({}).'.format(b, image.shape)

            mask[b[1]:b[3], b[0]:b[2], :] = 255

            mask_after = image_data_generator.random_transform(mask, seed=seed)[..., 0]
            mask_after = mask_after.copy()  # to force contiguous arrays

            # find bounding box again in augmented image
            [i, j] = np.where(mask_after == 255)
            if len(i)==0 or len(j)==0:
                print('Invalid augementation')
                wrongBox = True
                assert False
                continue
            boxes[index, 0] = float(min(j))
            boxes[index, 1] = float(min(i))
            boxes[index, 2] = float(max(j)) + 1 # set box to an open interval [min, max)
            boxes[index, 3] = float(max(i)) + 1 # set box to an open interval [min, max)

        if wrongBox:
            print('==== found a wrongbox! ====')
            wrongBox = False
            continue
        finished = True`

**Comment by hgaiser** [comment]
Does this solve your issue?

**Comment by hnsywangxin** [comment]
I have solved it ,thank you @hgaiser @de-vri-es @LaurensHagendoorn

**Comment by de-vri-es** [comment]
#190 may address this partially. There's still some open questions on how to deal with bounding boxes that get partially transformed out of the image canvas (see the inline comments). Any input there is welcome :)

Since the original issue is resolved, and a more general solution should be implemented in #190, I'm closing this issue :)
